### PR TITLE
 #167313414 search functionality

### PR DESCRIPTION
**Pull request by habinezadalvan**
#### What does this PR do?
- Creates a functionality for search the article.
#### Description of Task to be completed?
- Created the search functionality, and tests
#### How should this be manually tested?
- Clone  the repo
- create .env and add all the environment variables
- git checkout ft-Search-Functionality-1631414
- npm run migration
- npm run seeds
- npm run dev
- use postman and use get route and pass queries in the URL.; valid queries: 'title, keyword, author'
#### Any background context you want to provide?
- none
#### What are the relevant pivotal tracker stories?
[167313414](https://www.pivotaltracker.com/story/show/167313414)
#### Screenshots (if appropriate)
Screen Shot 2019-08-20 at 23 23 56

#### Questions:
none

### CODE REVIEW

**Review by salviosage — CHANGES_REQUESTED** [review]
consider other people's `req queries` whereby I can send offset and limit queries and it cant work the way you are validating queries.

File: src/services/article.service.js [hunk 1]
```diff
@@ -28,12 +28,13 @@ class articleService {
...
    * @static
+   * @param {*} query
    * @returns {object} data
    * @memberof articleService
    */
-  static async getAllArticles() {
+  static async getAllArticles(query = {}) {
     try {
-      return await db.findAll();
+      return await db.findAll({ where: query });
```
Comment (salviosage) [inline src/services/article.service.js]: each query can't work here like limit and offset. have to be out of `where`

File: src/middlewares/query.check.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      [Op.iLike]: `%${queries.title}%`
+    };
+  }
+
+  return queries;
+};
+
+export const checkQuery = (req, res, next) => {
+  const { query } = req;
+  const validQueries = ['author', 'keyword', 'tag', 'title', 'offset', 'limit'];
```
Comment (salviosage) [inline src/middlewares/query.check.js]: consider the other Queries might appear like `newest ` like `weekly` as we had that query in the previous UI we discussed.

**Review by salviosage — CHANGES_REQUESTED** [review]

**Comment by habinezadalvan** [comment]
> consider other people's `req queries` whereby I can send offset and limit queries and it cant work the way you are validating queries.

@salviosage I fixed that.

File: src/services/article.service.js (hunk 1 above)
Comment (habinezadalvan) [reply src/services/article.service.js]: It has been resolved

**Review by salviosage — APPROVED** [review]
LGTM

**Review by Mnickii — APPROVED** [review]
LGTM

**Review by minega25 — APPROVED** [review]

**Pull request merged**